src/platform_requirements.txt

```diff
@@ -1,2 +1,16 @@
-grpcio==1.38.0
+grpcio==1.62.2
```

Comment:
This doesn't seem right that we need to add so many platform_requirements.

Reply:
There are transitive dependencies which are C extensions, and were previously not declared.

Reply:
Ah.
So I don't always love the idea of pinning dependencies because it can confuse what we are actually interested in and makes it harder to upgrade the main package. Can we just leave these out?

Reply:
I would rather declare every single C dependency. Otherwise pipenv lock will choose those for us, and I have no assurance that whatever it chooses will be compatible with python 3.11.

Reply:
most of these are defined in the pipfile and you can just define them there.

Reply:
Hmmm...I'm not actually sure we need it.
@oliverchang WDYT?

Reply:
It's also surprising to me that we need to add this many. 

We package all our zips for all platforms (including Mac, windows) on Linux. this means that by default, we would be pulling all of the Linux versions of those C extensions and distributing them to Mac and windows. 

Can we validate that we actually need all of these C extensions? 

Re Pipfile vs platform_requirements: 

the platform_requirements.txt file was meant as a hacky override to "re-install" some of the results of the Pipfile resolution / install with platform-specific ones instead when are we are packaging up our  zips here: https://github.com/google/clusterfuzz/blob/7f39a19e5b87295ca9c3f14557eb3e82d7c0dc88/src/local/butler/package.py#L85

Reply:
We do need all these extensions, [proof](https://gist.github.com/vitorguidi/b2a6599e7f3c8bbfdf1741eb6525d571) (run pip install pipdeptree && pipdeptree so you can see dependency trees). 

The only margin we have to simplify this, is if the transitive C dependencies will be obtained from pip install on Mac and Windows. If that is actually the case, the only packages that should belong in platform_dependencies are the **C dependencies which are top level packages**.

If that is the case however, why do we not place psutil and pyyaml in platform_dependencies? How are these two specific packages able to propagate correctly if they are top leve, and not transitive, dependencies?

Reply:
Follow up on this: investigate what transitive C dependnecies are required on windows and Mac, and how they end up getting installed if we only pull grpcio/proto for multiplatform, and the rest is linux only

Reply:
+1.  Everything in the root Pipfile (https://github.com/google/clusterfuzz/blob/master/Pipfile) is expected to be installed on the system. This will certainly require C-extensions (and possibly transitively). 

src/Pipfile (which is separate from the root Pipfile) contains the packages that we download and re-package into zips for deployment. In theory, only grpcio and protobuf should require C extensions there, because we've never had any issue on Mac/Windows with just patching in the platform-specific versions of these two (which is what platform_requirements.txt does). 

Eventually, we do want to move everything from the root Pipfile into platform_requirements.txt, but perhaps we can do this more iteratively. e.g. just move a minimal seet that we have to absolutely upgrade for Python 3.11 support for now (e.g. psutil?)

Reply:
and perhaps with PYTHONPATH/sys.path ordering, we don't even have to uninstall the existing version of psutil on the system if the newer one we provide in the source zip takes precedence over the system one.

see https://github.com/google/clusterfuzz/blob/5d4bd65fa370a01b7d0006d404ca6c27fafa5e21/src/clusterfuzz/_internal/base/modules.py#L47 for how we mess with python module search paths on startup.

Reply:
Hmm there is a >0 chance that @jonathanmetzman https://github.com/google/clusterfuzz/pull/3663 was actually correct, and Windows only broke because it was not running psutil >= 5.9.4. The largest version I saw in our windows init scripts was 5.7.0, which is not compatible with py 3.11.

I might be wrong, but this seems very plausible

Reply:
I'm confused, i thought this was your theory. I thought i succeeded in upgrading the version on Windows.

Reply:
I was working under hte assumption that the update succeeded on linux but not on windows . Is that the case?